app/components/candidate_interface/application_visibility_component.rb

```diff
@@ -0,0 +1,20 @@
+class CandidateInterface::ApplicationVisibilityComponent < ViewComponent::Base
+  attr_reader :current_candidate, :application_form
+
+  def initialize(current_candidate:, application_form:)
+    @current_candidate = current_candidate
+    @application_form = application_form
+  end
+
+  def render?
+    FeatureFlag.active?(:candidate_preferences) && application_form.submitted_applications?
+  end
+
+  def pool_opt_in?
+    current_candidate.published_preferences&.last&.opt_in?
```

Comment:
```suggestion
    application_form.published_preference&.opt_in?
```

Let's use the application form rather then the candidate. This way it will work nicely with recruitment cycles